python/source/io.rst

```diff
@@ -497,3 +497,40 @@ the parquet file as :class:`ChunkedArray`
     pyarrow.Table
     col1: int64
     ChunkedArray = 0 .. 99
+
+Reading Line Delimited JSON
+===========================
+
+Arrow has builtin support for line-delimited JSON.
+Each line represents a row of data as a JSON object.
+
+Given some data in a file where each line is a JSON object
+containing a row of data:
+
+.. testcode::
+
+    import tempfile
+
+    with tempfile.NamedTemporaryFile(delete=False, mode="w+") as f:
+        f.write('{"a": 1, "b": 2.0, "c": 1}\n')
+        f.write('{"a": 3, "b": 3.0, "c": 2}\n')
+        f.write('{"a": 5, "b": 4.0, "c": 3}\n')
+        f.write('{"a": 7, "b": 5.0, "c": 4}\n')
```

Comment:
Should the examples sometimes show null entries? I see none of them in the rest of the cookbook.

Reply:
I think that when nulls change the behaviour of something and you need special code to deal with them we should have dedicated recipes. But in this case having nulls in the data shouldn't change how you deal with it, so I don't think there is much value in adding nulls

Reply:
Fair enough. I think it would be useful at some point to showcase null entries, just to show that they exist (silly, but some people may not be aware of that), but that's not necessarily in this section :-)